### ISSUE TITLE
Output "Ordering" for WTEXport

### ISSUE THREAD

**Issue by MTrop** [MTrop/DoomTools#3]
There is a need to have explicit control over texture ordering in the final list, since animations go by texture entry indexes between two searched names. For mods like BTSX, which rely on this engine behavior to add more animations to flats and textures and such between hardcoded animations, this is absolutely necessary.

### Proposal

An optional text file to provide that will describe the ordering of explicit entries. 

- Switch usage: `--ordering-textures [filename]`
- Switch usage: `--ordering-flats [filename]`
- If used, `--ordering-textures` _overrides_ the effects of `--null-texture`, replacing the set sorter completely.
- If used, `--ordering-flats` will impose a post-sort in the output WAD in the `FF_START/FF_END` namespace.
- This will also override the default behavior of figuring out what textures go into TEXTURE1 and TEXTURE2 via matching with the original set of textures, since the user will be able to specify exactly what potentially goes where.

Example format of the texture ordering input file:
```
[TEXTURE1]
AASHITTY
WFALL1
WFALL1A
WFALL1B
WFALL1C
WFALL1D
WFALL2
WFALL2A
WFALL2B
WFALL2C
WFALL2D
WFALL3
WFALL4

[TEXTURE2]
***OTHER
```

Only one `***OTHER` may exist between `TEXTURE1` and `TEXTURE2`. Leaving out a `TEXTURE2` section will make a blank one, if the base Wad had a `TEXTURE2`.

For flats:
```
***OTHER
SLIME01
SLIME01A
SLIME01B
SLIME02
SLIME03
SLIME04
```

### Feedback

Feedback is welcome. Texture manipulation in Doom sucks, but hopefully this will cover the rest of the bases.

**Comment by MTrop** [comment]
Some tweaks to the proposal:
The option switches are now `--ordering-file-tex` and `--ordering-file-flat`, the list sections are `:texture1` and `:texture2` to match the input list pattern, and "other" is `*other*`, and can only be placed in `:texture2` in the texture file, if the section exists.

**Comment by MTrop** [comment]
Yeah, cramming a lot of this in here at once is kinda ridiculous. All unmatched names will sort to the end of the relevant set of texture entries.

**Comment by MTrop** [comment]
Okay, _now_ this sounds like an over-engineered solution that is easily remedied by just removing the sort in `scanWAD(...)`.

I really should not have written this utility with what was probably several nights of little sleep, heh.

**Comment by MTrop** [comment]
I'm killing this issue. The root problem was solved in 2275a3b12ee8b7d1205b2e28b28ed9012ecb84e7 and is the best solution for 99% of use cases.
